wecs/panda3d/camera.py

```diff
@@ -13,48 +20,106 @@ class FirstPersonCamera:
     anchor_name: str = None
 
 
-# TODO:
-#   Rotate-around-character support
-#   Adjust distance so that the near plane is in front of level geometry
 @Component()
 class ThirdPersonCamera:
     camera: NodePath
     distance: float = 10.0
-    height: float = 3.0
-    focus_height: float = 2.0
-    dirty: bool = True
+    focus_height: float = 1.6
+
+
+@Component()
+class TurntableCamera:
+    heading : float = 0
+    pitch : float = 0
+    pivot: NodePath = field(default_factory=lambda:NodePath("camera pivot"))
+
+
+@Component()
+class CollisionZoom:
+    collision: CollisionNode = field(default_factory=lambda:CollisionNode("cam collisions"))
+    traverser: CollisionTraverser = field(default_factory=lambda:CollisionTraverser("cam traverser"))
+    queue: CollisionHandlerQueue = field(default_factory=lambda:CollisionHandlerQueue())
+    body_width: float = 0.5
 
 
 class UpdateCameras(System):
     entity_filters = {
-        '1stPerson': and_filter([
-            FirstPersonCamera,
-            Model,
-        ]),
-        '3rdPerson': and_filter([
+        '3rdPerson' : and_filter([
             ThirdPersonCamera,
-            Model,
+            Model
+
         ]),
+        '1stPerson' : and_filter([
+            FirstPersonCamera,
+            Model
+        ])
     }
 
     def init_entity(self, filter_name, entity):
         model = entity[Model]
-        if filter_name == '1stPerson':
+        if filter_name == "3rdPerson":
+            camera = entity[ThirdPersonCamera]
+            if TurntableCamera in entity:
+                turntable = entity[TurntableCamera]
+                turntable.pivot.reparent_to(render)
```

Comment:
`reparent_to(model.node)`, right? There's no guarantee that there even is a `render`.